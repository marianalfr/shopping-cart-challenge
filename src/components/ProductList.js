import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import ProductItem from './ProductItem';

// Styled Components -->

const TableHead = styled.div`
    position: relative;
    width: 100%;
    padding: 32px 0;
    display: flex;
    flex-flow: row nowrap;
    justify-content: space-between;
`;

const ColProduct = styled.div`
    width: 45%;
    display: flex;
    align-items: center;
    flex-flow: row nowrap;
`;

const Col = styled.div`
    width: 20%;
    display: flex;
    align-items: center;
    justify-content: center;
`;

const ColTotal = styled.div`
    width: 15%;
    display: flex;
    align-items: center;
    justify-content: center;
`;

const TitleTable = styled.h2`
    color: ${props => props.theme.color.light};
    text-transform: uppercase;
    letter-spacing: 1px;
    font-size: 10px;
    line-height: 16px;
`;

// ProductList Component ------------------------------------------------------------->

const ProductList = props => {
    return(
        <React.Fragment>
            <TableHead>
                <ColProduct>
                    <TitleTable>Product details</TitleTable>
                </ColProduct>
                <Col>
                    <TitleTable>Quantity</TitleTable>
                </Col>
                <Col>
                    <TitleTable>Price</TitleTable>
                </Col>
                <ColTotal>
                    <TitleTable>Total</TitleTable>
                </ColTotal>
            </TableHead>
            <ul>
                {props.shoppingCart.map(item => <li key={item.product.code}>
                        <ProductItem 
                            item = {item} 
                            updateQuantity = { props.updateQuantity }
                        />
                    </li>
                )}
            </ul>
        </React.Fragment>
    );
};

// ProductList.propTypes = {
//     shoppingCart: PropTypes.arrayOf(PropTypes.shape({
//         product: PropTypes.string,
//         price: PropTypes.number,
//         quantity: PropTypes.number,
//         code: PropTypes.string,
//         description: PropTypes.string,
//         images: PropTypes.shape({
//             thumb: PropTypes.string,
//             large: PropTypes.string
//         }),
//         offer: PropTypes.shape({
//             type: PropTypes.shape({
//                 category: PropTypes.string,
//                 name: PropTypes.string,
//                 numbers: PropTypes.shape({
//                     percentage: PropTypes.number,
//                     get: PropTypes.number,
//                     pay: PropTypes.number
//                 })
//             }),
//             minQty: PropTypes.number 
//         })
//     })),
//     updateQuantity: PropTypes.func
// };

export default ProductList;