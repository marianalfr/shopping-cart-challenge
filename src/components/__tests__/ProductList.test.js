import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import 'jest-styled-components';
import Adapter from 'enzyme-adapter-react-16';
import ProductList from '../ProductList';

Enzyme.configure({ adapter: new Adapter() });

const shoppingCart = [
  {
    product: {
      name: 'shirt',
      price: 20,
      code: 'X7R2OPX',
      description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.',
      images: {
        thumb: require('../../images/shirt.png'),
        large: require('../../images/shirt-large.jpg')
      },
      offer: {
        type: {
          category: 'percentage',
          name: '-5%',
          numbers: {
            percentage: 5
          }
        },
        minQty: 3 
      }
    },
    quantity: 3
  }
];

describe('ProductList Component', () => {

  it('renders correctly', () => {
    const wrapper = shallow(
      <ProductList 
          shoppingCart = { shoppingCart }
          updateQuantity = { () => {} }
      />
    );
    expect(wrapper.debug()).toMatchSnapshot(); 
  });
});