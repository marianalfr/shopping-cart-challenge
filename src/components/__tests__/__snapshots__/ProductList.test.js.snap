// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList Component renders correctly 1`] = `
"<Fragment>
  <styled.div>
    <styled.div>
      <styled.h2>
        Product details
      </styled.h2>
    </styled.div>
    <styled.div>
      <styled.h2>
        Quantity
      </styled.h2>
    </styled.div>
    <styled.div>
      <styled.h2>
        Price
      </styled.h2>
    </styled.div>
    <styled.div>
      <styled.h2>
        Total
      </styled.h2>
    </styled.div>
  </styled.div>
  <ul>
    <li>
      <ProductItem item={{...}} updateQuantity={[Function: updateQuantity]} />
    </li>
  </ul>
</Fragment>"
`;
